scripts/convert-bsp-specific-overrides

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+# Generalize overrides subsystem for NXP and Mainline support
+#
+# Essentially, we extend the overrides to a generic-bsp, nxp-bsp, and mainline-bsp.
+#
+# So, for example, the mx8mq override is split into:
+#
+# - imx-generic-bsp: compatible with every i.MX SoC and both BSP variants
+# - imx-nxp-bsp: compatible with every i.MX SoC but specific to NXP BSP
+# - imx-mainline-bsp: compatible with every i.MX SoC but specific to Mainline BSP
+#
+# - mx8-generic-bsp: compatible with every i.MX8 SoC and both BSP variants
+# - mx8-nxp-bsp: compatible with every i.MX8 SoC but specific to NXP BSP
+# - mx8-mainline-bsp: compatible with every i.MX8 SoC but specific to Mainline BSP
+#
+# - mx8m-generic-bsp: compatible with every i.MX8M SoC and both BSP variants
+# - mx8m-nxp-bsp: compatible with every i.MX8M SoC but specific to NXP BSP
+# - mx8m-mainline-bsp: compatible with every i.MX8M SoC but specific to Mainline BSP
+#
+# - mx8mq-generic-bsp: compatible with every i.MX8MQ SoC and both BSP variants
+# - mx8mq-nxp-bsp: compatible with every i.MX8MQ SoC8 but specific to NXP BSP
+# - mx8mq-mainline-bsp: compatible with every i.MX8MQ SoC but specific to Mainline BSP
+#
+# The extender mechanism is responsible for extending the override list to include the generic
+# overrides. We can then use the three different variants to handle the metadata correctly.
+#
+# Copyright 2022 (C) O.S. Systems Software LTDA.
+
+# Convert the recipes to use the new BSP-specific overrides.
+git ls-files \
+    | grep -L 'conf/machines/' \
+    | xargs sed -i \
+            -e 's,:\(mx[6-8]\w*\),:\1-nxp-bsp,g' \
+            -e 's,(\(mx[6-8]\w*\)),(\1-nxp-bsp),g' \
+            -e 's,\(mx[6-8]\w*\)|,\1-nxp-bsp|,g' \
+            -e 's,|\(mx[6-8]\w*\)),|\1-nxp-bsp),g' \
+            \
+            -e 's,:\(mx[5s]\w*\),:\1-generic-bsp,g' \
+            -e 's,(\(mx[5s]\w*\)),(\1-generic-bsp),g' \
+            -e 's,\(mx[5s]\w*\)|,\1-generic-bsp|,g' \
+            -e 's,|\(mx[5s]\w*\)),|\1-generic-bsp),g' \
+            \
+            -e 's,:\(vf\w*\),:\1-generic-bsp,g' \
+            -e 's,:\(vf[56]0\w*\),:\1-generic-bsp,g' \
+            -e 's,\(vf\w*\)|,\1-generic-bsp|,g' \
+            -e 's,|\(vf\w*\)),|\1-generic-bsp),g' \
+            -e 's,\(vf[56]0\w*\)|,\1-generic-bsp|,g' \
+            -e 's,|\(vf[56]0\w*\)),|\1-generic-bsp),g' \
+            \
+            -e 's,:\(imx\) ,:\1-nxp-bsp ,g' \
+            -e 's,(\(imx\)),(\1-nxp-bsp),g' \
+            -e 's,\(imx\)|,\1-nxp-bsp|,g' \
+            -e 's,|\(imx\)),|\1-nxp-bsp),g'
+
+for d in $(find -type d | egrep '/mx[6-8]w*'); do
```

Comment:
This section could have a comment that it is about converting override folder names to the new format.